iocage_lib/ioc_plugin.py

```diff
@@ -84,6 +84,125 @@ def __init__(self, release=None, jail=None, plugin=None, branch=None,
             # Backwards compat
             self.branch = 'master'
 
+    def clone_repo(self, depth=None):
+        if self.server == "download.freebsd.org":
+            git_server = "https://github.com/freenas/iocage-ix-plugins.git"
+        else:
+            git_server = self.server
+
+        git_working_dir = f"{self.iocroot}/.plugin_index"
+
+        if os.geteuid() == 0:
+            try:
+                self._clone_repo(
+                    self.branch, git_server, git_working_dir,
+                    depth, self.callback
+                )
+            except Exception as err:
+                iocage_lib.ioc_common.logit(
+                    {
+                        "level": "EXCEPTION",
+                        "message": err
+                    },
+                    _callback=self.callback
+                )
+
+    @staticmethod
+    def fetch_plugin_packagesites(package_sites):
+        def download_parse_packagesite(packagesite_url):
+            package_site_data = {}
+            try:
+                response = requests.get(f'{packagesite_url}/All/', timeout=20)
+                response.raise_for_status()
+
+                for pkg in re.findall(r'<a.*>\s*(\S+).txz</a>', response.text):
```

Comment:
Please use `html.parser` module instead

Reply:
I did, it's way slower

Reply:
I've got it wrong. Let's use `etree.HTMLParser` from `lxml`

Reply:
We cant add yet another dependency to iocage only for this

Reply:
If lxml is not a dependency of iocage then fine, let's leave it like this @sonicaj